envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/EnvoyListenersFactory.kt

```diff
@@ -0,0 +1,298 @@
+    private val egressFilters: List<HttpFilterFactory> = listOf()
+) {
+    private val accessLogTimeFormat = stringValue(listenersFactoryProperties.httpFilters.accessLog.timeFormat)
+    private val accessLogMessageFormat = stringValue(listenersFactoryProperties.httpFilters.accessLog.messageFormat)
+    private val accessLogLevel = stringValue(listenersFactoryProperties.httpFilters.accessLog.level)
+    private val accessLogLogger = stringValue(listenersFactoryProperties.httpFilters.accessLog.logger)
+
+    companion object {
+        private const val egressRdsInitialFetchTimeout: Long = 20
+        private const val ingressRdsInitialFetchTimeout: Long = 30
+
+        val defaultServiceTagFilterRules = serviceTagFilterRules()
+        val defaultHeaderToMetadataConfig = headerToMetadataConfig(defaultServiceTagFilterRules)
+        val defaultHeaderToMetadataFilter = { _: Group -> headerToMetadataHttpFilter(defaultHeaderToMetadataConfig) }
+        val defaultEnvoyRouterHttpFilter = { _: Group -> envoyRouterHttpFilter() }
+        val defaultApiConfigSource: ApiConfigSource = apiConfigSource()
+        val defaultEgressFilters = listOf(defaultHeaderToMetadataFilter, defaultEnvoyRouterHttpFilter)
+        val defaultIngressFilters = listOf(defaultEnvoyRouterHttpFilter)
+
+        private fun envoyRouterHttpFilter(): HttpFilter = HttpFilter.newBuilder().setName("envoy.router").build()
+
+        fun apiConfigSource(): ApiConfigSource {
+            return ApiConfigSource.newBuilder()
+                    .setApiType(ApiConfigSource.ApiType.GRPC)
+                    .addGrpcServices(GrpcService.newBuilder()
+                            .setEnvoyGrpc(
+                                    GrpcService.EnvoyGrpc.newBuilder()
+                                            .setClusterName("envoy-control-xds")
+                            )
+                    ).build()
+        }
+
+        fun headerToMetadataHttpFilter(headerToMetadataConfig: HeaderToMetadataConfig.Builder): HttpFilter {
+            return HttpFilter.newBuilder()
+                    .setName("envoy.filters.http.header_to_metadata")
+                    .setTypedConfig(ProtobufAny.pack(
+                            headerToMetadataConfig.build()
+                    ))
+                    .build()
+        }
+
+        fun headerToMetadataConfig(rules: List<HeaderToMetadataConfig.Rule>): HeaderToMetadataConfig.Builder {
+            val headerToMetadataConfig = HeaderToMetadataConfig.newBuilder()
+                    .addRequestRules(
+                            HeaderToMetadataConfig.Rule.newBuilder()
+                                    .setHeader("x-canary")
+                                    .setRemove(false)
+                                    .setOnHeaderPresent(
+                                            HeaderToMetadataConfig.KeyValuePair.newBuilder()
+                                                    .setKey("canary")
```

Comment:
we should use `CanaryProperties.metadataKey` property here

(or we can create a separate task for making canary and service-tags consistent. It will also require changes in docs, where we write now that you have to configure static config properly, that info will be outdated now when we merge this)